src/Microsoft.Data/DataFrame.IO.cs

```diff
@@ -0,0 +1,261 @@
+                        BaseColumn intColumn = new PrimitiveColumn<int>(names[i]);
+                        df.InsertColumn(i, intColumn);
+                        break;
+                    case Type longType when longType == typeof(long):
+                        BaseColumn longColumn = new PrimitiveColumn<long>(names[i]);
+                        df.InsertColumn(i, longColumn);
+                        break;
+                    case Type sbyteType when sbyteType == typeof(sbyte):
+                        BaseColumn sbyteColumn = new PrimitiveColumn<sbyte>(names[i]);
+                        df.InsertColumn(i, sbyteColumn);
+                        break;
+                    case Type shortType when shortType == typeof(short):
+                        BaseColumn shortColumn = new PrimitiveColumn<short>(names[i]);
+                        df.InsertColumn(i, shortColumn);
+                        break;
+                    case Type uintType when uintType == typeof(uint):
+                        BaseColumn uintColumn = new PrimitiveColumn<uint>(names[i]);
+                        df.InsertColumn(i, uintColumn);
+                        break;
+                    case Type ulongType when ulongType == typeof(ulong):
+                        BaseColumn ulongColumn = new PrimitiveColumn<ulong>(names[i]);
+                        df.InsertColumn(i, ulongColumn);
+                        break;
+                    case Type ushortType when ushortType == typeof(ushort):
+                        BaseColumn ushortColumn = new PrimitiveColumn<ushort>(names[i]);
+                        df.InsertColumn(i, ushortColumn);
+                        break;
+                    case Type stringType when stringType == typeof(string):
+                        BaseColumn stringColumn = new StringColumn(names[i], 0);
+                        df.InsertColumn(i, stringColumn);
+                        break;
+                    default:
+                        throw new NotSupportedException();
+                }
+            }
+
+            // Fills values.
+            using (var st = createStream())
+            {
+                string line = st.ReadLine();
+                long nbline = 0;
+                rowline = 0;
+                while (line != null && (nrows == -1 || rowline < nrows))
+                {
+                    var spl = line.Split(sep);
+                    if (header && nbline == 0)
+                    {
+                        // Skips.
+                    }
+                    else
+                    {
+                        df.AppendRow(rowline, spl);
+                        ++rowline;
+                    }
+                    ++nbline;
+                    line = st.ReadLine();
+                }
+            }
+
+            if (index)
+            {
+                // TODO: Bug in InsertColumn. Can only insert at the end. Should support inserting in the middle
```

Comment:
What's the plan for handling this TODO?

Reply:
Filed https://github.com/dotnet/corefxlab/issues/2695